src/openrct2/ride/ride.c

```diff
@@ -4710,8 +4710,12 @@ static void vehicle_create_trains(sint32 rideIndex, sint32 x, sint32 y, sint32 z
     }
 
     // Link the first train and last train together
-    firstTrain.head->prev_vehicle_on_ride = lastTrain.tail->sprite_index;
-    lastTrain.tail->next_vehicle_on_ride = firstTrain.head->sprite_index;
+    if(firstTrain.head != NULL && lastTrain.tail != NULL) {
```

Comment:
This all should be extracted to a separate PR with `openrct2_assert(firstTrain.head != NULL, "Invalid train");` or some other helpful message, for both trains.

Reply:
I think this blip can be removed from the PR. It was related to MinGW in release seeing different errors, which was causing Travis errors

Reply:
Yeah, it should still be addressed though. See https://github.com/OpenRCT2/OpenRCT2/pull/6443#issuecomment-336146711 about triggering it in other builds, maybe we can have one of the Arch Travis builds use `-fno-PIC` (it would need `-O2` or `-O3` too)?